### ISSUE TITLE
Increase precision to microsecond when using `time` 0.3.x

### ISSUE THREAD

**Issue by yerke** [blackbeam/rust_mysql_common#68]
Hi @blackbeam.

Thank you very much for maintaining this crate.

Is it possible to increase precision of `time::PrimitiveDateTime`, `time::Time` to microsecond when using `time` 0.3.x?

As far as I understand `time` supports up to nanosecond precision, and so we just need to provide the necessary formatting.

`time = { version = "0.3.11", features = ["serde", "parsing", "formatting", "local-offset", "serde-human-readable", "macros"] }`

```rust
#[cfg(test)]
mod tests {
    use time::{format_description, PrimitiveDateTime};
    use time::macros::datetime;

    #[test]
    fn test_time() {
        let format = format_description::parse("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond]").unwrap();
        let datetime = "2020-01-02 03:04:05.123456";
        let parsed = PrimitiveDateTime::parse(datetime, &format).unwrap();
        assert_eq!(parsed, datetime!(2020-01-02 03:04:05.123456000));
        println!("{parsed:?}");

        let formatted = parsed.format(&format).unwrap();
        println!("{formatted}");

        assert_eq!(datetime, formatted);
    }
}
```

There is also https://time-rs.github.io/api/time/format_description/modifier/enum.SubsecondDigits.html if it's somehow helpful.

Thanks!

Edit: I just found that there a modifier for a strict number of digits in subsecond: `let format = format_description::parse("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond digits:6]").unwrap();`.
https://github.com/time-rs/time/discussions/418

**Comment by blackbeam** [comment]
Published as [v0.29.1](https://github.com/blackbeam/rust_mysql_common/releases/tag/v0.29.1)

**Comment by yerke** [comment]
Thank you for such quick turnaround!
